python/cudf/cudf/core/index.py

```diff
@@ -1616,6 +1616,40 @@ def quarter(self):
         res = extract_quarter(self._values)
         return Int8Index(res, dtype="int8")
 
+    def isocalendar(self):
+        """
+        Returns a DataFrame with the year, week, and day
+        calculated according to the ISO 8601 standard.
+        Returns
+        -------
+        DataFrame
+            with columns year, week and day
+        Examples
+        --------
+
+        >>> gIndex = cudf.DatetimeIndex(["2020-05-31 08:00:00",
+        ...    "1999-12-31 18:40:00"])
+        >>> gIndex.isocalendar()
+                             year  week  day
+        2020-05-31 08:00:00  2020    22    7
+        1999-12-31 18:40:00  1999    52    5
+        """
+        indexSeries = cudf.core.tools.datetimes.to_iso_calendar(self)
+
+        @property
+        def day(self):
+            return indexSeries["day"]
+
+        @property
+        def week(self):
+            return indexSeries["week"]
+
+        @property
+        def year(self):
+            return indexSeries["year"]
+
+        return indexSeries
```

Comment:
```suggestion
        return cudf.core.tools.datetimes.to_iso_calendar(self)
```

I don't think we need to define properties inside this method as they can be accessed from the result dataframe too like..:
```python
>>> gIndex.isocalendar().week # this will fetch the `week` column of the dataframe.
```

Reply:
You're right! Not sure why I didn't think of that. Thanks, Prem!